### PR TITLE
Homepage hero and nav

### PR DESCRIPTION
**Pull request by drewclem**
## TLDR
- Mobile and Desktop header/navigation
- HomepageHero
- Interior page type
- vClickOutside

### Mobile and Desktop header/navigation
- Due to the layout on desktop, I broke up the navigation into two sections: main and legal.
- Mobile loops through both sections to generate a single list with appropriate stylings
- Desktop loops through each section separately placing main on the left of the logo and legal on the right
- This is our first instance of building out a submenu for navigation. I found that the BaseMenu component, while it does give accessiblity features, it's not flexible to being styled differently for mobile and desktop so it's been removed for a simpler navigation menu

### HomepageHero
- Fairly straight forward. This component takes a heading, subheading (optional), button (optional), and image.
- It has the option to change the background image, but defaults to the baseball crowd thing.
- It has substantial padding and margin involved to account for the absolute positioned header

### Interior page type
- This is a new pattern we're going to start introducing. It allows us to give a little more guardrails to clients as they build interior pages because it will have its own schema in which we can require certain components
- It essentially does the same thing as the Page component both in storyblok and the codebase, but with the additional requirement of an InteriorTitle component.
- The InteriorTitle component will be required because of the design spec with the absolutely positioned header

### vClickOutside
- I noticed our menus/modals/etc were missing a clickoutside event.
- Basically, when a modal, submenu, etc pops open- the user expectation is that clicking outside of it will close it.
- I've added the vClickOutside library, and you can inspect `BaseModal` to see how it works- but it simplifies what we have to do to trigger a close event when someone clicks outside of an open pop up of some sort

### CODE REVIEW

**Review by redevdesign — APPROVED** [review]
It all looks so good!!!
